### ISSUE TITLE
XML parsing: Content between tags is lost partially using function `to_xml`

### ISSUE THREAD

**Issue by Kopfbremse** [mikefarah/yq#1447]
**Version of yq**: 4.30.4
**Operating system**: linux
**Installed via**: binary release

**Bug description**
XML parsing: Content between tags is lost partially using function `to_xml`

**Input**
```yaml
date: "2022-11-25"
raw:
  id: 1
  XML: text_outside_1<Tag1 />text_outside_2<Tag2>Kling Klong</Tag2>text_outside_3
  time: 09:19:00
```

**Command**
```bash
yq '.raw.XML |= from_xml'
```

**Output**
```yaml
date: "2022-11-25"
raw:
  id: 1
  XML:
    +content: text_outside_3
    Tag1:
    Tag2: Kling Klong
  time: 09:19:00
```

**Unexpected behaviour**
- 'text_outside_1' and 'text_outside_2' are dropped
- 'text_outside_3' is converted.

----------

**Expected output (version 1)**
```yaml
date: "2022-11-25"
raw:
  id: 1
  XML:
    +content: text_outside_1
    Tag1:
    +content: text_outside_2
    Tag2: Kling Klong
    +content: text_outside_3
  time: 09:19:00
```
----------

**Expected output (version 2)**
```yaml
date: "2022-11-25"
raw:
  id: 1
  XML:
    +content_1: text_outside_1
    Tag1:
    +content_2: text_outside_2
    Tag2: Kling Klong
    +content_3: text_outside_3
  time: 09:19:00
```
----------

**Expected output (version 3)**
```yaml
date: "2022-11-25"
raw:
  id: 1
  XML:
    +content: text_outside_1\ntext_outside_2\ntext_outside_3
    Tag1:
    Tag2: Kling Klong
  time: 09:19:00
```
Add to documentation, that content between sibling tags is joined

----------

**Expected output (version 4)**
```yaml
date: "2022-11-25"
raw:
  id: 1
  XML:
    +content:
      - text_outside_1
      - text_outside_2
      - text_outside_3
    Tag1:
    Tag2: Kling Klong
  time: 09:19:00
```
Add to documentation, that content between sibling tags is converted as array

----------

**Expected output (version 5)**
```yaml
date: "2022-11-25"
raw:
  id: 1
  XML:
    Tag1:
    Tag2: Kling Klong
  time: 09:19:00
```
Add to documentation, that content between tags is not supported.

**Comment by mikefarah** [comment]
Nice pick up, will fix in next release

**Comment by mikefarah** [comment]
Fixed in 4.30.5

**Comment by Kopfbremse** [comment]
The fixed version "4.30.5" now generated an error for XML that contains tags with mixed content.
This is a bit sad, because it is [valid](https://www.w3schools.com/xml/schema_complex_mixed.asp) to define XML this way.
At least the error message `Error: invalid XML: Encountered chardata [text_outside_1] outside of XML node` is misleading. It should be something like `XML contains tags with mixed content. This is currently not supported by yq`.

**Comment by Kopfbremse** [comment]
Processing of XML tags with mixed content is now possible ... thanks a lot !
